2017-10-27  Diana Nguyen Hoang  <classdho@lnxsrv07.seas.ucla.edu>

        * timespec.h: changed the timespec_cmp function	by expanding
        the comparison explained about the function to try and fix the
        overflow.
--- timespec.h	2017-10-26 23:33:30.249780000 -0700
+++ mytimespec.h	2017-10-26 23:59:13.102777000 -0700

@@ -45,8 +45,23 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  int  diff = a.tv_sec - b.tv_sec;
+  //return diff ? diff : a.tv_nsec - b.tv_nsec;
+  //if diff < 0 return -1 
+  //if diff == 0 return 0
+  //if diff > 0 return 1
+  if (diff < 0)
+    {
+      return -1; 
+    }
+  if (diff == 0)
+    {
+      return 0; 
+    }
+  if (diff > 0)
+    {
+      return 1; 
+    }
 }
 
 # if ! HAVE_DECL_NANOSLEEP
